src/App.tsx

```diff
@@ -1,223 +1,293 @@
+      setDeletingTodosIds(prev => prev.filter(id => id !== todoId));
+    }
   }, []);
 
+  const deleteCompleated = () => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        removeTodo(todo.id);
+      }
+    });
+  };
+
+  const onUpdateTodo = (chosenTodo: Todo) => {
+    setTodosToUpdate(prev => [...prev, chosenTodo]);
+
+    updateTodo(chosenTodo.id, chosenTodo.title, chosenTodo.completed)
+      .then((udpatedTodo) => {
+        setTodos(currentTodos => currentTodos.map(todo => (
+          todo.id !== udpatedTodo.id
+            ? todo
+            : {
+              id: udpatedTodo.id,
+              userId: udpatedTodo.userId,
+              title: udpatedTodo.title,
+              completed: udpatedTodo.completed,
+            }
+        )));
+      })
+      .catch(() => {
+        setErrorMessage(ErrorTypes.UnableToUpdate);
+      })
+      .finally(() => {
+        setTodosToUpdate([]);
+      });
+  };
+
+  const onToggleAll = () => {
+    setIsLoading(true);
+    if (isToggledAll === false) {
+      todos.forEach(todo => {
+        if (!todo.completed) {
+          onUpdateTodo({ ...todo, completed: !todo.completed });
+        }
+      });
+      setIsToggledAll(true);
+    } else {
+      todos.forEach(todo => {
+        onUpdateTodo({ ...todo, completed: !todo.completed });
+      });
+      setIsToggledAll(false);
+    }
+
+    setIsLoading(false);
+  };
+
+  const visibleTodos = useMemo(() => {
+    switch (filterType) {
+      case FilterTypes.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+
+      case FilterTypes.COMPLETED:
+        return todos.filter(todo => todo.completed);
+
+      case FilterTypes.All:
+        return todos;
+
+      default:
+        throw new Error('Invalid sorting type');
+    }
+  }, [todos, filterType]);
+
+  if (isError) {
+    setTimeout(() => {
+      setIsError(false);
+    }, 3000);
+  }
+
+  const handleCloseErrorMessage = () => {
+    setIsError(false);
+  };
+
+  const incompletedTodos = visibleTodos.filter(todo => !todo.completed);
+  const completedTodosAmount = visibleTodos
+    .filter(todo => todo.completed).length;
+  const toggledAlltodos = visibleTodos.every(todo => todo.completed);
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
+        <Header
+          newTodoField={newTodoField}
+          isLoading={isLoading}
+          onAddTodo={addNewTodo}
+          onToggleAll={onToggleAll}
+          toggledAlltodos={toggledAlltodos}
```

Comment:
What is the different between these 2 props `onToggleAll ` vs `toggledAlltodos `

Reply:
onToggleAll - function that toggled all todos
toggledAlltodos - constant to check if all todos are toggled